crates/bin/pcli/src/command/tx/auction/dutch.rs

```diff
@@ -190,87 +196,145 @@ impl DutchCmd {
             }
             DutchCmd::DutchAuctionEnd {
                 all,
-                auction_id,
+                auction_ids,
                 source,
+                batch,
                 fee_tier,
             } => {
-                let auction_ids = match (all, auction_id) {
+                let auction_ids = match (all, auction_ids.is_empty()) {
                     (true, _) => auctions_to_end(app.view(), *source).await?,
-                    (false, Some(auction_id)) => {
-                        let auction_id = auction_id.parse::<AuctionId>()?;
-                        vec![auction_id]
-                    }
-                    (false, None) => {
-                        bail!("auction_id is required when --all is not set")
+                    (false, false) => auction_ids.to_vec(),
+                    (false, true) => {
+                        bail!("auction_ids are required when --all is not set")
                     }
                 };
 
-                let mut planner = Planner::new(OsRng);
+                if auction_ids.is_empty() {
+                    println!("no active auctions to end");
+                    return Ok(());
+                }
 
-                planner
-                    .set_gas_prices(gas_prices)
-                    .set_fee_tier((*fee_tier).into());
+                // Process auctions in batches
+                let batches = auction_ids.chunks(*batch as usize);
+                for (batch_num, auction_batch) in batches.clone().enumerate() {
+                    let auctions_progress_count: usize = if batch_num == 0 {
+                        auction_batch.len()
+                    } else {
+                        auction_batch.len() + (batch_num as usize * *batch as usize)
+                    };
+                    println!(
+                        "processing batch {} of {}, containing auctions {} of {} ...",
+                        batch_num + 1,
+                        batches.len(),
+                        auctions_progress_count,
+                        auction_ids.len()
+                    );
```

Comment:
```suggestion
                    println!(
                        "processing batch {} of {}, starting with {}",
                        batch_num + 1,
                        batches.len(),
                        batch_num * *batch as usize
                    );
```

Seems better to avoid adding too much logic.